generator/internal/genclient/translator/protobuf/protobuf_test.go

```diff
@@ -329,60 +240,69 @@ message Fake {
 				Typez:         genclient.BYTES_TYPE,
 			},
 		},
+		OneOfs: []*genclient.OneOf{
```

Comment:
This is a bit strange.  I get that Protobuf chooses to represent optional fields as one-of fields with a single branch. Do we need to use that representation in our own data structures?

Reply:
So the reason this is important in proto is because it is safe to move optinal into a one-of without a breaking change, because of how this is implemented by protobuf. That is the reason I kept optionals as one-ofs. As written for Rust, this would be a breaking change which I think is fine as long as we know we are making this choice. Other languages, like Go, also implement oneofs in a way where we can't freely move in and out of oneofs. That said other langauges may chose differently than we have with Rust. That is why I left in this information. But if you think we should simpify this in our model and say only explicit oneofs are oneofs I am fine with that, there are just tradeoffs.

Reply:
Hmmm, I see your point. I will remove optionals from one-of and say compatibility as an exercise to the impl of a given language.